_posts/2025-07-15-fraud-detection-e2e.md

```diff
@@ -0,0 +1,597 @@
+
+By structuring the inference step this way, the pipeline ensures that the deployed model always uses the freshest features for each prediction, supporting robust, real-time fraud detection.
+
+## Importing and Running the Pipeline
+
+Once your environment is set up and the data is uploaded, you're ready to run the pipeline.
+
+### Import the Pipeline
+
+1. Open the Kubeflow Pipelines UI (usually at [http://localhost:8080](http://localhost:8080) if you used the default port-forward).
+2. Click **Pipelines** in the sidebar, then click **Upload pipeline**.
+3. Upload the compiled pipeline YAML file (e.g., `pipeline/fraud-detection-e2e.yaml`).
+
+### Run the Pipeline
+
+1. After uploading, click on your pipeline in the list.
+2. Click **Create run**.
+3. Fill in the run details and click **Start**.
+
+You can monitor the progress and view logs for each step directly in the UI.
+
+## Testing the Live Endpoint
+
+With the inference service running, you can now interact with your deployed model in real time. Let's see how to send prediction requests and interpret the results.
+
+Before sending requests, port-forward the inference pod so the service is accessible locally. **Run this command in a separate terminal window:**
+
+```sh
+kubectl -n kubeflow get pods -l component=predictor -o jsonpath="{.items[*].metadata.name}" | tr ' ' '\n' | grep '^fraud-detection' | head -n1 | xargs -I {} kubectl port-forward -n kubeflow pod/{} 8081:8080
+```
+
+With the port-forward active, you can now send a request to the model:
+
+```sh
+curl -X POST http://localhost:8081/v1/models/onnx-model:predict \
+-H "Content-Type: application/json" \
+-d '{"user_id": "user_0"}'
+```
+
+The service retrieves features for `user_0`, runs a prediction, and returns the fraud probability.
+
+```json
+{"user_id":"user_0","prediction":[[0.8173668384552002]]}
+```
+
+> Note: The result of the prediction may vary depending on the initial raw data you uploaded.  
+> Try sending requests with a few different `user_id` values (e.g., `"user_1"`, `"user_2"`, etc.) to see how the predictions change.
+
+## Conclusion
+
+This post has walked you through a complete, reproducible AI/ML workflow—from raw data to a live model serving endpoint—using Kubeflow and open source tools. Along the way, you've seen how to prepare data with Spark, manage features with Feast, train and register models, and deploy real-time inference services with KServe, all orchestrated in a portable pipeline you can run on your own laptop.
+
+By following this blueprint, you can adapt and extend the process for your own machine learning projects, whether you're working locally or scaling up to production. Kubeflow's modular platform and ecosystem make it possible to manage the entire ML lifecycle in a consistent, automated, and open way.
+
+Ready to try it yourself? The complete source code for this project is available on [GitHub](https://github.com/hbelmiro/fraud_detection_e2e_demo/tree/kubeflow).
```

Comment:
in an ideal world we'd put this in the example repos but we've stated that the examples repo is not maintained.